### PR TITLE
Dependency injection

### PR DESCRIPTION
**Pull request by andriy1507**
Injected ViewModel for SplashActivity using Dagger.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -4,12 +4,14 @@ apply plugin: 'kotlin-android'
...
 
+apply plugin: 'kotlin-kapt'
+
 android {
     compileSdkVersion targetSdkVersion
     buildToolsVersion toolsVer
     defaultConfig {
         applicationId appId
-        minSdkVersion minSdkVersion
+        minSdkVersion minimalSdkVersion
```
Comment (EugeneKryvoshey) [inline app/build.gradle:14]: Please revert this change.

File: app/src/main/java/com/goryachok/forecastapp/WeatherApplication.kt [hunk 2]
```diff
@@ -2,13 +2,17 @@ package com.goryachok.forecastapp
...
+import com.goryachok.forecastapp.dagger.splashViewModel.DaggerViewModelComponent
+import com.goryachok.forecastapp.dagger.splashViewModel.ViewModelComponent
 import com.goryachok.forecastapp.network.api.OpenWeatherMapAPI
 import com.goryachok.forecastapp.network.connectivity.ConnectivityInterceptor
 import com.goryachok.forecastapp.network.data.PREFS_NAME
 import com.goryachok.forecastapp.network.data.WeatherDataRepository
 
 class WeatherApplication : Application() {
 
+    val component: ViewModelComponent = DaggerViewModelComponent.create()
```
Comment (EugeneKryvoshey) [inline app/src/main/java/com/goryachok/forecastapp/WeatherApplication.kt:14]: Please use lazy initialization.

File: app/src/main/java/com/goryachok/forecastapp/WeatherApplication.kt [hunk 3]
```diff
@@ -2,13 +2,17 @@ package com.goryachok.forecastapp
...
 import android.app.Application
 import android.content.Context
+import com.goryachok.forecastapp.dagger.splashViewModel.DaggerViewModelComponent
+import com.goryachok.forecastapp.dagger.splashViewModel.ViewModelComponent
 import com.goryachok.forecastapp.network.api.OpenWeatherMapAPI
 import com.goryachok.forecastapp.network.connectivity.ConnectivityInterceptor
 import com.goryachok.forecastapp.network.data.PREFS_NAME
 import com.goryachok.forecastapp.network.data.WeatherDataRepository
 
 class WeatherApplication : Application() {
```
Comment (EugeneKryvoshey) [inline app/src/main/java/com/goryachok/forecastapp/WeatherApplication.kt:12]: Please, create interface App() and inherit Application class from it. Extract 'component' as interface member.

File: app/src/main/java/com/goryachok/forecastapp/activities/SplashActivity.kt [hunk 4]
```diff
@@ -13,23 +13,29 @@ import androidx.core.content.ContextCompat
...
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
+import javax.inject.Inject
 
-class StartActivity : AppCompatActivity() {
+class SplashActivity : AppCompatActivity() {
+
+    @Inject lateinit var viewModel: SplashViewModel
 
     override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (EugeneKryvoshey) [inline app/src/main/java/com/goryachok/forecastapp/activities/SplashActivity.kt:26]: Please create BaseActivity() class which has protected abstract fun setupDependencies() within onCreate() callback.

File: app/src/main/java/com/goryachok/forecastapp/dagger/splashViewModel/ViewModelComponent.kt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package com.goryachok.forecastapp.dagger.splashViewModel
+
+import com.goryachok.forecastapp.activities.SplashActivity
+import com.goryachok.forecastapp.viewmodel.SplashViewModel
+import dagger.Component
+
+@Component(modules = [ViewModelModule::class])
+interface ViewModelComponent {
+    fun getViewModel(): SplashViewModel
```
Comment (EugeneKryvoshey) [inline app/src/main/java/com/goryachok/forecastapp/dagger/splashViewModel/ViewModelComponent.kt:9]: Please remove this.

**Review by EugeneKryvoshey — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (andriy1507) [reply app/build.gradle:14]: Gradle takes value from DefaultProductFlavor.class so I changed name to minSdkVer

File: app/src/main/java/com/goryachok/forecastapp/WeatherApplication.kt (hunk 2 above)
Comment (andriy1507) [reply app/src/main/java/com/goryachok/forecastapp/WeatherApplication.kt:14]: Fixed

File: app/src/main/java/com/goryachok/forecastapp/WeatherApplication.kt (hunk 3 above)
Comment (andriy1507) [reply app/src/main/java/com/goryachok/forecastapp/WeatherApplication.kt:12]: Done

File: app/src/main/java/com/goryachok/forecastapp/activities/SplashActivity.kt (hunk 4 above)
Comment (andriy1507) [reply app/src/main/java/com/goryachok/forecastapp/activities/SplashActivity.kt:26]: Done

File: app/src/main/java/com/goryachok/forecastapp/dagger/splashViewModel/ViewModelComponent.kt (hunk 5 above)
Comment (andriy1507) [reply app/src/main/java/com/goryachok/forecastapp/dagger/splashViewModel/ViewModelComponent.kt:9]: Removed
